SwiftDraft/ReliantFramework/ReliantFrameworkTests/ReliantFrameworkTests.swift

```diff
@@ -54,17 +54,25 @@ struct ContextNeedingContext : ReliantContext {
         return ContextNeedingContext()
     }
 }
+class SubWaver : Waver {
+    func wave(reason: String) -> String {
+        return "Substitute waving"
+    }
+}
 
-
-
-
+class SubstituteContext : SimpleReferenceContext {
+    override init() {
+        super.init()
+        waver = SubWaver()
```

Comment:
Can't we work through protocols and delegation instead?
Eg: protocol Waver, structs RealWaver and SubWaver, SubWaver initialises RealWaver and delegates to it when needed, overriding it's behaviour when wanted?